docs/content/guides/dagster/pythonic-resources.mdx

```diff
@@ -0,0 +1,440 @@
+        ),
+    },
+)
+```
+
+If we instead would like the configuration for our credentials to be provided at runtime, we can use the `configure_at_launch()` method to defer the construction of the `CredentialsResource` until runtime.
+
+Because `credentials` requires runtime configuration through the launchpad, it must also be passed to the `Definitions` object, so that configuration can be provided at runtime. Nested resources only need to be passed to the `Definitions` object if they require runtime configuration.
+
+```python file=/guides/dagster/pythonic_resources/pythonic_resources.py startafter=start_new_resource_dep_job_runtime endbefore=end_new_resource_dep_job_runtime
+credentials = CredentialsResource.configure_at_launch()
+
+
+defs = Definitions(
+    assets=...,
+    resources={
+        "credentials": credentials,
+        "bucket": FileStoreBucket(
+            credentials=credentials,
+            region="us-east-1",
+        ),
+    },
+)
+```
+
+---
+
+### Testing configurable resources
+
+You can test the initialization of a <PyObject object="ConfigurableResource"/> by constructing it manually. In most cases, you can construct your resource directly:
+
+```python file=/guides/dagster/pythonic_resources/pythonic_resources.py startafter=start_new_resource_testing endbefore=end_new_resource_testing
+from dagster._config.structured_config import ConfigurableResource, ConfigurableResourceAdapter
+
+V = TypeVar("V")
+
+
+class ResourceDependency(Generic[V]):
+    def __get__(self, obj: "Resource", __owner: Any) -> V:
+        return getattr(obj, self._name)
+
+
+class MyResource(ConfigurableResource):
+    value: str
+
+    def get_value(self) -> str:
+        return self.value
+
+
+def test_my_resource():
+    assert MyResource(value="foo").get_value() == "foo"
+```
+
+If your resource requires other resources, then you can pass them as constructor arguments.
+
+```python file=/guides/dagster/pythonic_resources/pythonic_resources.py startafter=start_new_resource_testing_with_context endbefore=end_new_resource_testing_with_context
+from dagster._config.structured_config import ConfigurableResource
+
+
+class StringHolderResource(ConfigurableResource):
+    value: str
+
+
+class MyResourceRequiresAnother(ConfigurableResource):
+    foo: StringHolderResource
+    bar: str
+
+
+def test_my_resource_with_context():
+    string_holder = StringHolderResource(value="foo")
+    resource = MyResourceRequiresAnother(foo=string_holder, bar="bar")
+    assert resource.foo.value == "foo"
+    assert resource.bar == "bar"
+```
+
+---
+
+## Defining Pythonic IO managers
+
+Pythonic [IO managers](/concepts/io-management/io-managers) are defined as subclasses of <PyObject object="ConfigurableIOManager"/>, and similarly to Pythonic resources specify any configuration fields as attributes. Each subclass must implement a `handle_output` and `load_input` method, which are called by the Dagster runtime to handle the storing and loading of data.
```

Comment:
```suggestion
Pythonic [I/O managers](/concepts/io-management/io-managers) are defined as subclasses of <PyObject object="ConfigurableIOManager"/>, and similarly to Pythonic resources specify any configuration fields as attributes. Each subclass must implement a `handle_output` and `load_input` method, which are called by the Dagster runtime to handle the storing and loading of data.
```

Also:

> ... by the Dagster runtime to handle the storing and loading of data.

Is `the Dagster runtime` a thing? (I'm not trying to sound snarky, I honestly don't know - it read a bit like a typo)